cassandra/io/libevreactor.py

```diff
@@ -126,8 +126,11 @@ def _cleanup(self):
                     watcher.stop()
 
         self.notify()  # wake the timer watcher
-        log.debug("Waiting for event loop thread to join...")
-        self._thread.join(timeout=1.0)
+        # PYTHON-752 Thread might have just been created and not started
+        try:
+            self._thread.join(timeout=1.0)
+        except RuntimeError:
+            pass
```

Comment:
I worry a little about catching an error as general as `RuntimeError` -- would it be a good idea to check the message content and re-raising if it's not what we're looking for? I don't love it, but I worry about ignoring something important that subclasses `RuntimeError`.

Reply:
Hmm, yeah, what do you think of adding a debug inside the except like `log.warning(e.message)` or similar?

Reply:
I don't have a strong opinion either way on logging, but I don't think it solves the problem I'm worried about. To be clear, I meant something like:

```python
try:
    self._thread.join(timeout=1.0)
except RuntimeError:
    if 'cannot join thread before it is started' not in e.message:
        raise
```

What are your thoughts on that?

Reply:
Yeah agree, I don't like too much as well examining the error message. Let me see if I can think of better solution, if nothing comes up I think we can leave it like this.

Reply:
I think I'll go with this, I'd rather not add a lock for a thread that's run once at the very end of the execution.

Reply:
I don't think I object to the lock as much as you do, but works for me.